beater/beater.go

```diff
@@ -107,15 +120,66 @@ func (bt *beater) Run(b *beat.Beat) error {
 		return nil
 	}
 
-	bt.server = newServer(bt.config, pub.Send)
+	bt.server = newServer(bt.config, tracer, pub.Send)
 	bt.mutex.Unlock()
 
-	err = run(bt.server, lis, bt.config)
-	if err == http.ErrServerClosed {
-		bt.logger.Infof("Listener stopped: %s", err.Error())
-		return nil
+	var g errgroup.Group
+	g.Go(func() error {
+		return run(bt.server, lis, bt.config)
+	})
+	if bt.config.Tracing.isEnabled() {
+		g.Go(func() error {
+			return bt.server.Serve(traceListener)
+		})
+	}
+	if err := g.Wait(); err != http.ErrServerClosed {
+		return err
+	}
+	bt.logger.Infof("Server stopped")
+	return nil
+}
+
+// initTracer configures and returns an elasticapm.Tracer for tracing
+// the APM server's own execution.
+func initTracer(info beat.Info, config *Config, logger *logp.Logger) (*elasticapm.Tracer, net.Listener, error) {
+	if !config.Tracing.isEnabled() {
+		os.Setenv("ELASTIC_APM_ACTIVE", "false")
```

Comment:
would be nice if everything that is passed via env vars could be passed as well as an argument to `NewTracer`, since we are importing the agent already; but it is fine

Reply:
Agreed, I have an issue for this: https://github.com/elastic/apm-agent-go/issues/88